R/FilterState.R

```diff
@@ -86,7 +87,7 @@ FilterState <- R6::R6Class( # nolint
       if (length(extract_type) == 1 && is.null(dataname)) {
         stop("if extract_type is specified, dataname must also be specified")
       }
-
+      private$id <- sample.int(.Machine$integer.max, 1)
```

Comment:
I think having the id inside the class is ok as an object can know "who it is". I think the id should be passed in from FilterStates however - and only created here if no id is passed in

Reply:
Yup, this is exactly what I'm trying to do now:
- remove FilterState$id
- implement `id` in `FilterStates$set_filter_state` where new `FilterState` will have attribute attached in `state_list`